onadata/apps/api/viewsets/openid_connect_viewset.py

```diff
@@ -126,8 +126,10 @@ def callback(self, request, **kwargs):  # pylint: disable=no-self-use
                     return Response(
                         data, template_name='missing_oidc_detail.html')
 
-                first_name = claim_values.get(FIRST_NAME)
                 last_name = claim_values.get(LAST_NAME)
+                first_name = claim_values.get(FIRST_NAME)
+                if not first_name:
+                    first_name = last_name
```

Comment:
What about the case when we do not have the `last_name` and the `first_name`?

Reply:
Mhh I'll implement functionality to fail the authentication process(_I think this was the agreed upon approach ?_) if both fields are missing.

Reply:
Yes.